### PR TITLE
Genio 2 recipe

### PR DESCRIPTION
**Pull request by jackburton79**
Recipe for Genio 2.

Release information available here: 
https://github.com/Genio-The-Haiku-IDE/Genio/releases/tag/v2

### CODE REVIEW

**Comment by Begasus** [comment]
Trailing whitespace on line 56.
Could you also delete the older recipe?

File: haiku-apps/genio/genio-2.0.0.recipe [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+PROVIDES="
+	genio${secondaryArchSuffix} = $portVersion
+	app:Genio = $portVersion
+	"
+REQUIRES="
+	haiku${secondaryArchSuffix}
+	gcc_syslibs${secondaryArchSuffix}_devel
+	llvm17${secondaryArchSuffix}_clang
+	lib:libgit2${secondaryArchSuffix}
+	lib:libyaml_cpp${secondaryArchSuffix}
```
Comment (Begasus) [inline haiku-apps/genio/genio-2.0.0.recipe:28]: These brackets are not needed here for lib:lib* (same for devel:lib*) and cmd:gcc
```diff
-	lib:libgit2${secondaryArchSuffix}
-	lib:libyaml_cpp${secondaryArchSuffix}
+	lib:libgit2$secondaryArchSuffix
+	lib:libyaml_cpp$secondaryArchSuffix
```

File: haiku-apps/genio/genio-2.0.0.recipe [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libgit2${secondaryArchSuffix} >= 1.5
+	devel:libyaml_cpp${secondaryArchSuffix} >= 0.8
+	"
+BUILD_PREREQUIRES="
+	makefile_engine
+	cmd:gcc${secondaryArchSuffix}
```
Comment (Begasus) [inline haiku-apps/genio/genio-2.0.0.recipe:38]: No {} needed here:
```diff
-	cmd:gcc${secondaryArchSuffix}
+	cmd:gcc$secondaryArchSuffix
```

**Review by Begasus — CHANGES_REQUESTED** [review]

**Comment by jackburton79** [comment]
Can't apply these changes until this evening, probably.

File: haiku-apps/genio/genio-2.0.0.recipe (hunk 1 above)
Comment (Begasus) [reply haiku-apps/genio/genio-2.0.0.recipe:28]: In cases where you can't use something like `devel:libfoo$secondaryArchSuffix` but need to revert to the provides in the package like `libfoo${secondaryArchSuffix}_devel`
For instance you could use `llvm17${secondaryArchSuffix}_devel`, but the package provides also `devel:libLLVM_17$secondaryArchSuffix` which is prefered.

Comment (jackburton79) [reply haiku-apps/genio/genio-2.0.0.recipe]: done

File: haiku-apps/genio/genio-2.0.0.recipe (hunk 2 above)
Comment (jackburton79) [reply haiku-apps/genio/genio-2.0.0.recipe]: done

File: haiku-apps/genio/genio-2.0.0.recipe [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
 COPYRIGHT="2022-2023 The Genio Team"
 LICENSE="BSD (3-clause)
 	MIT"
 REVISION="1"
-srcGitRev="0655bca6b6851e746db160f949b831955580b337"
-SOURCE_URI="https://github.com/Genio-The-Haiku-IDE/Genio/archive/v1.tar.gz"
-CHECKSUM_SHA256="7ae9a8affd59869741b7a891384fc733ce78fd404ba2e433708a68a473d11358"
-SOURCE_FILENAME="Genio-v1.tar.gz"
-SOURCE_DIR="Genio-1"
+srcGitRev="eae9cf7a34c099a9dfb579aa0a478bf5e7840836"
```
Comment (Begasus) [inline haiku-apps/genio/genio-2.0.0.recipe:10]: srcGitRev can be removed here, then it should be good to go :)
```diff
-srcGitRev="eae9cf7a34c099a9dfb579aa0a478bf5e7840836"
```

File: haiku-apps/genio/genio-2.0.0.recipe [hunk 4]
```diff
@@ -1,17 +1,17 @@
 SUMMARY="The Haiku IDE"
-DESCRIPTION="Genio is an IDE for the Haiku operating system. \
+DESCRIPTION="Genio is a native IDE for the Haiku operating system. \
 It has a rich editor supporting syntax highlight, auto completion and other advanced features. \
```
Comment (humdingerb) [inline haiku-apps/genio/genio-2.0.0.recipe:3]:
```suggestion
It has a rich editor supporting syntax highlighting, auto completion and other advanced features. \
```

**Review by humdingerb — COMMENTED** [review]
Tried the recipe on 32bit nightly. Works.

File: haiku-apps/genio/genio-2.0.0.recipe (hunk 3 above)
[the remark this answers is not in the document]
Comment (jackburton79) [reply haiku-apps/genio/genio-2.0.0.recipe]: done

File: haiku-apps/genio/genio-2.0.0.recipe [hunk 5]
```diff
@@ -21,15 +20,21 @@ PROVIDES="
 	app:Genio = $portVersion
 	"
 REQUIRES="
-	haiku$secondaryArchSuffix
+	haiku${secondaryArchSuffix}
```
Comment (Begasus) [inline haiku-apps/genio/genio-2.0.0.recipe:23]: Nitpicking here, can be changed on next version:
```diff
-	haiku${secondaryArchSuffix}
+	haiku$secondaryArchSuffix
```

**Comment by Begasus** [comment]
Thanks! 👍

**Pull request merged**